ui/src/app/layout/staging/staging.component.ts

```diff
@@ -71,9 +76,22 @@ export class StagingComponent implements OnInit {
     async prepareStaging(): Promise<any> {
         this.staging = null;
         this.stagingResult = [];
-        return this.environmentsService.prepareStaging(this.selectedApplication.id, this.selectedEnvironment).then(s => {
+        return this.environmentsService.prepareStaging(this.selectedApplication.id, this.selectedEnvironment).then(async s => {
+
             this.staging = s;
             this.changes = s.changes.map(change => ({ change: change, selected: true }));
+
+            const producerRemovedChange = this.changes.filter(change => change.change.changeType === 'TOPIC_PRODUCER_APPLICATION_REMOVED');
+            if (producerRemovedChange.length) {
+                this.mappedDeleteProducerIds = await Promise.all(
+                    producerRemovedChange[0].change.topicProducerIds.map(ids => this.applicationInfo(ids)));
+            }
+
+            const producerAddChange = this.changes.filter(change => change.change.changeType === 'TOPIC_PRODUCER_APPLICATION_ADDED');
+            if (producerAddChange.length) {
+                this.mappedAddProducerIds = await Promise.all(
+                    producerAddChange[0].change.topicProducerIds.map(ids => this.applicationInfo(ids)));
```

Comment:
Great code, really :-) But, if you make the suggested change of having only ONE application per change, this will get much simpler...